src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    private String color;
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public abstract double getArea();
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public void draw() {
+        moreInfo();
+        System.out.println("Area: " + getArea() + " sq. units");
+        System.out.println("Color: " + color);
+        System.out.println();
+    }

```

Comment:
The `draw()` method should not be implemented in the abstract class if it's meant to have different behaviors in different subclasses. It's better to make it abstract and implement it in each subclass, or if there's common behavior, consider using an interface with a default method.